logger/mocklogger.go

```diff
@@ -0,0 +1,200 @@
+package logger
+
+import (
+	"io"
+)
+
+// MockRouteCountLogger is a mock implementation of KayveeLogger that counts the router rules
+// applied to each log call without actually formatting or writing the log line.
+type MockRouteCountLogger struct {
+	logger      *Logger
+	routeCounts map[string]int
+}
+
+// GetRuleCounts returns a map of rule names to the number of times that rule has been applied
+// in routing logs for MockRouteCountLogger. Only includes routing rules that have at least
+// one use.
+func (ml *MockRouteCountLogger) GetRuleCounts() map[string]int {
+	out := make(map[string]int)
+	for k, v := range ml.routeCounts {
+		out[k] = v
+	}
+	return out
+}
+
+// NewMockCountLogger returns a new MockRoutCountLogger with the specified `source`.
+func NewMockCountLogger(source string) *MockRouteCountLogger {
+	return NewMockCountLoggerWithContext(source, nil)
+}
+
+// NewMockCountLoggerWithContext returns a new MockRoutCountLogger with the specified `source` and `contextValues`.
+func NewMockCountLoggerWithContext(source string, contextValues map[string]interface{}) *MockRouteCountLogger {
+	routeCounts := make(map[string]int)
+	lg := NewWithContext(source, contextValues)
+	lg.fLogger = &routeCountingFormatLogger{
+		routeCounts: routeCounts,
+	}
+	mocklg := MockRouteCountLogger{
+		logger:      lg,
+		routeCounts: routeCounts,
+	}
+	return &mocklg
+}
+
+/////////////////////////////
+//
+// routeCountingFormatLogger
+//
+/////////////////////////////
+
+// routeCountingFormatLogger implements the formatLogger interface to allow for counting
+// invocations of routing rules.
+type routeCountingFormatLogger struct {
+	routeCounts map[string]int
+}
+
+// formatAndLog tracks routing statistics for this mock router.
+// Initialization works as with the default format logger, but no formatting or logging is actually performed.
+func (fl *routeCountingFormatLogger) formatAndLog(data map[string]interface{}) {
+	routeData, ok := data["_kvmeta"]
+	if !ok {
+		return
+	}
+	routes, ok := routeData.(map[string]interface{})["routes"]
+	if !ok {
+		return
+	}
+	for _, route := range routes.([]map[string]interface{}) {
+		rule := route["rule"].(string)
+		fl.routeCounts[rule] = fl.routeCounts[rule] + 1
+	}
+}
+
+// setFormatter implements the FormatLogger method.
+func (fl *routeCountingFormatLogger) setFormatter(formatter Formatter) {
+	// we don't format anything in this mock logger
```

Comment:
If they're being called but just don't need to be implemented, this makes sense as is